product/roundhouse.databases.mysql/parser/Parser.cs

```diff
@@ -0,0 +1,160 @@
+        }
+
+        public bool AnsiQuotes
+        {
+            get
+            {
+                return this.ansiQuotes;
+            }
+            set 
+            {
+                this.ansiQuotes = value;
+
+                if (scanner != null) {
+                    scanner.AnsiQuotes = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Parses the MySQL script and returns a List of ParsedStatement
+        /// instances.
+        /// </summary>
+        /// <returns>List of ParsedStatement</returns>
+        public List<ParsedStatement> Parse() {
+
+            // initialize our starting location
+            start = 0;
+            current = 0;
+
+            tokens = scanner.Scan();
+
+            while (!IsAtEnd()) {
+                start = current;
+                statements.Add(ParseStatement());
+            }
+
+            return statements;
+        }
+
+        private ParsedStatement ParseStatement() 
+        {
+            Token token = Advance();
+            string delimiter = ";";
+            bool setDelimiter = false;
+            ParsedStatement.Type statementType = ParsedStatement.Type.Sql;
+
+            while (!IsAtEnd() && !token.TokenType.Equals(Token.Type.Delimiter)) {
+
+                switch(token.TokenType) {
+
+                    case Token.Type.EndOfFile:
+                        break;
+
+                    case Token.Type.DelimiterDeclare:
+                        statementType = ParsedStatement.Type.Delimiter;
+                        setDelimiter = true;
+                        break;
+
+                    case Token.Type.Delimiter:
+                        if (setDelimiter) {
+                            // the delimiter will not be part of the statemtent's value
+                            delimiter = token.Value;
+                        }
+                        break;
+                }
+
+                token = Advance();
+            }
+
+            StringBuilder builder = new StringBuilder();
+            foreach (Token tokenThis in tokens.GetRange(start, (current - start))) {
+
+                if(tokenThis.TokenType.Equals(Token.Type.EndOfLine)) {
+                    builder.Append(Environment.NewLine);
+                } if(tokenThis.TokenType.Equals(Token.Type.Delimiter)) {
+                    // the end of the statement doesn't require a delimiter
+                } else {
+                    builder.Append(tokenThis.Value);
+                }
+            }
+
+            return new ParsedStatement(statementType, builder.ToString(), delimiter);
+        }
+
+        private Token Advance()
+        {
+            current++;
+            return tokens[current - 1];
+        }
+
+        private bool IsAtEnd()
+        {
+            bool value = false;
+
+            if (current >= tokens.Count) {
+                value = true;
+            }
+
+            return value;
+        }
+    }
```

Comment:
This is nitpicking, I'm sorry, but I really would simplify this to

```suggestion
        private bool IsAtEnd()
        {
            return current >= tokens.Count;
        }
```